package/system/signify/patches/0004-Allow-compilation-of-VERIFY_ONLY-1-without-libbsd.patch

```diff
@@ -0,0 +1,254 @@
++ *    notice, this list of conditions and the following disclaimer.
++ * 2. Redistributions in binary form must reproduce the above copyright
++ *    notice, this list of conditions and the following disclaimer in the
++ *    documentation and/or other materials provided with the distribution.
++ * 3. The name of the author may not be used to endorse or promote products
++ *    derived from this software without specific prior written permission.
++ *
++ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
++ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
++ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL
++ * THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
++ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
++ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
++ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
++ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
++ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
++ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
++ */
++
++/*
++ * Rejected in glibc
++ * <https://sourceware.org/ml/libc-alpha/2006-03/msg00125.html>.
++ */
++
++#include <errno.h>
++#include <string.h>
++#include <stdlib.h>
++
++#if defined(_WIN32) || defined(__WIN32__) || defined(__WINDOWS__)
++#define LIBBSD_IS_PATHNAME_SEPARATOR(c) ((c) == '/' || (c) == '\\')
++#else
++#define LIBBSD_IS_PATHNAME_SEPARATOR(c) ((c) == '/')
++#endif
++
++#ifdef HAVE___PROGNAME
++extern const char *__progname;
++#else
++static const char *__progname = NULL;
++#endif
++
++const char *
++getprogname(void)
++{
++#if defined(HAVE_PROGRAM_INVOCATION_SHORT_NAME)
++	if (__progname == NULL)
++		__progname = program_invocation_short_name;
++#elif defined(HAVE_GETEXECNAME)
++	/* getexecname(3) returns an absolute pathname, normalize it. */
++	if (__progname == NULL)
++		setprogname(getexecname());
++#endif
++
++	return __progname;
++}
+diff --git a/sha2.c b/sha2.c
+index da609c6..e27c82c 100644
+--- a/sha2.c
++++ b/sha2.c
+@@ -36,7 +36,12 @@
+ 
+ #include <sys/types.h>
+ 
++#ifndef VERIFYONLY
+ #include <bsd/string.h>
++#else
++#include <string.h>
++#endif
++
+ #include "sha2.h"
+ 
+ /*
+diff --git a/signify.c b/signify.c
+index c4369fd..2ecfd98 100644
+--- a/signify.c
++++ b/signify.c
+@@ -22,7 +22,13 @@
+ #include <limits.h>
+ #include <stdint.h>
+ #include <fcntl.h>
++
++#ifndef VERIFYONLY
+ #include <bsd/string.h>
++#else
++#include <string.h>
++#endif
++
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <stddef.h>
+@@ -30,9 +36,16 @@
+ #include <err.h>
+ #include <unistd.h>
+ #include <errno.h>
++
++#ifndef VERIFYONLY
+ #include <bsd/readpassphrase.h>
+ #include <bsd/libutil.h>
+ #include <bsd/stdlib.h>
++#else
++const char* getprogname(void);
++size_t strlcpy(char *dst, const char *src, size_t dsize);
```

Comment:
`strlcpy()` seems like it would be orthogonal... I think you can declare it and not use it without the compiler puking...